diff -rupN wec_map/ext_emconf.php wec_map_new/ext_emconf.php
--- wec_map/ext_emconf.php	2009-12-29 11:18:33.000000000 +0100
+++ wec_map_new/ext_emconf.php	2009-12-29 11:16:29.000000000 +0100
@@ -3,7 +3,7 @@
 ########################################################################
 # Extension Manager/Repository config file for ext "wec_map".
 #
-# Auto generated 29-12-2009 11:18
+# Auto generated 26-12-2009 14:24
 #
 # Manual updates:
 # Only the data in the array - everything else is removed by next
diff -rupN wec_map/ext_tables.php wec_map_new/ext_tables.php
--- wec_map/ext_tables.php	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/ext_tables.php	2009-12-29 11:16:29.000000000 +0100
@@ -91,6 +91,13 @@ if(t3lib_extMgm::isLoaded('tt_address'))
 			'zip' => 'zip',
 			'country' => 'country',
 		),
+                'filter' => array(
+                        'groupTable' => 'tt_address_group', // table to filter by
+                        'groupTableNameField' => 'title', //field with the 'title' for each item in the group table
+                        'isMM' => true, // [true|false] are there any mm table?
+                        'MMtable' => 'tt_address_group_mm', //if 'isMM' is true then 'MMtable' should be set
+                        'groupField' =>'', //if 'isMM' is false this should be set with then name of the field that have a comma sepparated values to filter by
+                    ),
 	);
 }
 
diff -rupN wec_map/locallang_db.xml wec_map_new/locallang_db.xml
--- wec_map/locallang_db.xml	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/locallang_db.xml	2009-12-29 11:16:29.000000000 +0100
@@ -89,6 +89,7 @@
       <label index="tt_content.pi_flexform.showRadiusSearch">Show Radius Search</label>
       <label index="tt_content.pi_flexform.kml">KML to overlay</label>
       <label index="tt_content.pi_flexform.showSidebar">Show Marker Sidebar</label>
+      <label index="tt_content.pi_flexform.showFilterbar">Show Marker Filterbar</label>
       <label index="tt_content.pi_flexform.prefillAddress">Prefill Address from User Info</label>
     </languageKey>
     <languageKey index="de" type="array">
diff -rupN wec_map/pi3/class.tx_wecmap_pi3.php wec_map_new/pi3/class.tx_wecmap_pi3.php
--- wec_map/pi3/class.tx_wecmap_pi3.php	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/pi3/class.tx_wecmap_pi3.php	2009-12-29 11:16:29.000000000 +0100
@@ -141,6 +141,9 @@ class tx_wecmap_pi3 extends tslib_pibase
 		$showSidebar = $this->pi_getFFvalue($piFlexForm, 'showSidebar', 'mapConfig');
 		empty($showSidebar) ? $showSidebar = $conf['showSidebar']:null;
 		$this->showSidebar = $showSidebar;
+
+                $showFilterbar = $this->pi_getFFvalue($piFlexForm, 'showFilterbar', 'mapConfig');
+		empty($showSidebar) ? $showSidebar = $conf['showFilterbar']:null;
 		
 		$tables = $this->pi_getFFvalue($piFlexForm, 'tables', 'default');
 		empty($tables) ? $tables = $conf['tables']:null;
@@ -245,6 +248,8 @@ class tx_wecmap_pi3 extends tslib_pibase
 			}
 		}
 		
+                $filterby = intval($this->piVars['filterby']);                
+
 		// there are two ways of buiding the SQL query:
 		// 1. from the data given via flexform
 		// 2. all manually from TS
@@ -260,8 +265,21 @@ class tx_wecmap_pi3 extends tslib_pibase
 				}
 				
 				$where .= $this->cObj->enableFields($table);
-				
-				$res = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows('*', $table, $where);
+
+                                // if the filterby parameter is set the query should by modified to get only selected markers
+                                $extraTable = '';
+                                if($filterby){
+                                    if ($GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['isMM']){
+                                        // if there is a true MM table
+                                        $extraTable .= ','.$GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['MMtable'];
+                                        $where .= ' AND uid=uid_local AND uid_foreign='.$filterby;
+                                    }else{
+                                        // there isnt any mm table
+                                        $where .= tx_wecmap_shared::listQueryFromCSV($GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['groupField'], $filterby, $table, 'OR');;
+                                    }
+                                }
+
+				$res = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows('*', $table.$extraTable, $where);
 
 				foreach( $res as $key => $data ) {
 					$conf['table'] = $table;
@@ -304,7 +322,19 @@ class tx_wecmap_pi3 extends tslib_pibase
 				
 				$where .= $this->cObj->enableFields($table);
 
-				$res = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows($select, $from, $where);
+                                $extraTable = '';
+                                if($filterby){
+                                    if ($GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['isMM']){
+                                        // if there is a true MM table
+                                        $extraTable .= ','.$GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['MMtable'];
+                                        $where .= ' AND uid=uid_local AND uid_foreign='.$filterby;
+                                    }else{
+                                        // there isnt any mm table
+                                        $where .= tx_wecmap_shared::listQueryFromCSV($GLOBALS['TCA'][$table]['ctrl']['EXT']['wec_map']['filter']['groupField'], $filterby, $table, 'OR');;
+                                    }
+                                }
+
+				$res = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows($select, $from.$extraTable, $where);
 
 				// add icon if configured, else see if we just have an iconID
 				// and use that. We assume the icon is added somewhere else.
@@ -337,6 +367,8 @@ class tx_wecmap_pi3 extends tslib_pibase
 		if($showRadiusSearch) $content['addressForm'] = $this->getAddressForm();
 		if($showWrittenDirs) $content['directions'] = $this->getDirections();
 		if($showSidebar) $content['sidebar'] = $this->getSidebar();
+                // show filterbar only it is enabled
+                if($showFilterbar) $content['filterbar'] = $this->getFilterbar($pid,$GLOBALS['TCA'][$conf['table']]['ctrl']['EXT']['wec_map']['filter']['groupTable'],$GLOBALS['TCA'][$conf['table']]['ctrl']['EXT']['wec_map']['filter']);
 
 		// run all the content pieces through TS to assemble them
 		$output = tx_wecmap_shared::render($content, $conf['output.']);
@@ -468,6 +500,39 @@ class tx_wecmap_pi3 extends tslib_pibase
 
 		return $out;
 	}
+
+        /**
+         * Return a list similar to the sidebar to filter markers by group
+         * $pid integer
+         * $table string table name that has the groups
+         * $conf array filter metadata configuration
+         */
+        function getFilterbar($pid,$table,$conf){
+                $filterItems=''; //each item to filter by
+                if(!empty($pid)) {
+                        $where = '1=1' . tx_wecmap_shared::listQueryFromCSV('pid', $pid, $table, 'OR');
+                } else {
+                        $where = '1=1';
+                }
+
+                $where .= $this->cObj->enableFields($table);
+
+                // gets the rows of the group table
+                $res = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows('*', $table, $where);
+                foreach( $res as $key => $data ) {
+                        $linkParams = array(
+                            //URL parameter as filter
+                            $this->prefixId.'[filterby]' => $data['uid'],
+                            );
+                        //the link itself
+                        $data['filter_link'] = $this->pi_linkTP($data[$conf['groupTableNameField']],$linkParams);
+                        $filterItems .= tx_wecmap_shared::render($data, $this->conf['filterbarItem.']);
+                        
+                }
+                //wrap all filter items inside the filterbar TS
+                $out = tx_wecmap_shared::render(array('map_height' => $this->height, 'map_id' => $this->mapName, 'content' => $filterItems), $this->conf['filterbar.']);
+		return $out;
+        }
 }
 
 
diff -rupN wec_map/pi3/flexform_ds.xml wec_map_new/pi3/flexform_ds.xml
--- wec_map/pi3/flexform_ds.xml	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/pi3/flexform_ds.xml	2009-12-29 11:16:29.000000000 +0100
@@ -141,6 +141,14 @@
 							</config>
 						</TCEforms>
 					</showSidebar>
+                                        <showFilterbar>
+						<TCEforms>
+							<label>LLL:EXT:wec_map/locallang_db.xml:tt_content.pi_flexform.showFilterbar</label>
+							<config>
+								<type>check</type>
+							</config>
+						</TCEforms>
+					</showFilterbar>
 					<prefillAddress>
 						<TCEforms>
 							<label>LLL:EXT:wec_map/locallang_db.xml:tt_content.pi_flexform.prefillAddress</label>
diff -rupN wec_map/pi3/static/setup.txt wec_map_new/pi3/static/setup.txt
--- wec_map/pi3/static/setup.txt	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/pi3/static/setup.txt	2009-12-29 11:16:29.000000000 +0100
@@ -68,8 +68,10 @@ plugin.tx_wecmap_pi3 {
 	defaultdescription 	< tx_wecmap_api.defaultdescription
 	homeicon    		< tx_wecmap_api.homeicon
 	sidebar 			< tx_wecmap_api.sidebar
+        filterbar 			< tx_wecmap_api.filterbar
 	directions 			< tx_wecmap_api.directions
 	sidebarItem 		< tx_wecmap_api.sidebarItem
+        filterbarItem 		< tx_wecmap_api.filterbarItem
 	addressForm 		< tx_wecmap_api.addressForm
 	output 				< tx_wecmap_api.output
 	_CSS_DEFAULT_STYLE 	< tx_wecmap_api._CSS_DEFAULT_STYLE
diff -rupN wec_map/static/setup.txt wec_map_new/static/setup.txt
--- wec_map/static/setup.txt	2009-12-29 11:18:32.000000000 +0100
+++ wec_map_new/static/setup.txt	2009-12-29 11:16:29.000000000 +0100
@@ -45,7 +45,25 @@ tx_wecmap_api {
 	  10.insertData = true
 	  10.if.isTrue.field = info_title
 	}
-	
+
+        filterbarItem = COA
+        filterbarItem {
+	  10 = TEXT
+	  10.field = filter_link
+	  10.wrap = <li>|</li>
+	  10.insertData = true
+	  10.if.isTrue.field = filter_link
+	}
+
+        filterbar = COA
+	filterbar {
+		10 = TEXT
+		10.field = content
+		10.wrap = <div id="{field:map_id}_filterbar" class="filterbar" style="height:{field:map_height}px;"><ul>|</ul></div>
+		10.insertData = true
+                10.if.isTrue.field = content
+	}
+
 	output = COA
 	output {
 		20       = HTML
@@ -53,6 +71,9 @@ tx_wecmap_api {
 		30       = TEXT
 		30.field = map
 		30.wrap  = <td>|</td>
+                35       = TEXT
+		35.field = filterbar
+                35.wrap  = <td style="vertical-align: top;">|</td>
 		40       = TEXT
 		40.field = sidebar
 		40.wrap  = <td style="vertical-align: top;">|</td>
